src/lms-harmonizer/edfi_lms_harmonizer/helpers/argparser.py

```diff
@@ -202,20 +224,12 @@ def parse_main_arguments(args_in: List[str]) -> MainArguments:
         args_parsed.engine,
         args_parsed.server,
         args_parsed.dbname,
+        args_parsed.username,
+        args_parsed.password,
         args_parsed.port,
         args_parsed.encrypt,
         args_parsed.trust_certificate,
+        args_parsed.useintegratedsecurity
     )
 
-    if arguments.engine == DB_ENGINE.MSSQL:
-        if args_parsed.useintegratedsecurity:
-            arguments.build_mssql_adapter_with_integrated_security()
-        else:
-            arguments.build_mssql_adapter(
-                args_parsed.username,
-                args_parsed.password,
-            )
-    else:
-        arguments.build_postgresql_adapter(args_parsed.username, args_parsed.password)
-
```

Comment:
This older code was creating the adapter as soon as the arguments are parsed. I wanted to throw an exception if username and password are missing. However, the logging is not setup (in `__main__`) until _after_ argument parsing. Thus I moved the adapter initialization into the `get_adapter` function for lazy loading.